docs/backup-recovery/README.md

```diff
@@ -0,0 +1,132 @@
+
+
+### On the hub ###
+
+- User creates a TALO CR that defines:
+  - A set of clusters to be upgraded  
+  - The need for resource backup
+- User applies the TALO CR to the hub cluster at the beginning of the maintenance window
+- TALO checks the backup requirement in the TALO CR. If required, deploys a backup workload on the designated spoke.  
+
+### States ###
+
+- BackupStatePreparingToStart - is the initial state all clusters are automatically assigned to on the first reconciliation pass of the TALO CR. Upon entry TALO deletes spoke backup namespace and hub view resources that might have remained from the prior incomplete attempts.
+- BackupStateStarting - is the state for creation of the backup job pre-requisites and the job itself
+- BackupStateActive - the job is in "Active" state
+- BackupStateSucceeded - a final state reached when the backup job has succeeded
+- BackupStateTimeout - a final state meaning that artifact backup has been partially done
+- BackupStateError - a final state reached when the job ends with a non-zero exit code
+
+### On the spoke ###
+
+The backup workload generates an utility called `upgrade-recovery.sh` and pass the `--take-backup` option to trigger backup of `/etc`, `/usr/local`, `/var/lib/kubelet` directories and `etcd database` in the recovery partition or at the recovery folder at `/var/recovery`. In addition, the active OS deployment is pinned using ostree and the standby deployments are removed.
+
+#### Procedure end options ####
+
+- Success (“Completed”)
+- Failure due to timeout (“DeadlineExceeded”)  
+  - the procedure did not complete due to a timeout and can be continued
+  - Unrecoverable error (“BackoffLimitExceeded”) - user intervention is desired to determine the failure reason
+
+
+## Recovery from Upgrade Failure
+
+### Platform Rollback
+
+Platform rollback, if needed, is handled by issuing the `rpm-ostree rollback -r` command. Not all upgrades will include
+an update to the platform OS, however, so this step will not always be required.
+
+When the backup was taken, the active deployment was pinned and standby deployments removed. Use the
+`ostree admin status` command to see the current state.
+
+|Deployment Status|Rollback Action|
+|-----------------|---------------|
+|Single deployment, pinned|Platform OS unchanged, no action required|
+|Standby deployment pinned, flagged as "rollback"|Trigger rollback with `rpm-ostree rollback -r` command|
+|Multiple standby deployments, pinned deployment is not "rollback"|1. Delete unpinned standby deployments with `ostree admin undeploy 1`, until pinned deployment is "rollback".<br>2. Trigger rollback with `rpm-ostree rollback -r` command|
+
+### Recovery Utility
+
+The upgrade recovery utility is generated when taking the backup, before the upgrade starts, written as `/var/recovery/upgrade-recovery.sh`.
+
+The first phase of the recovery will run the following steps, and then stop to allow the user to reboot the node:
```

Comment:
I think we can make the description of the two stages more generalized, as well, removing implementation details. If someone is interested, they can look at the script implementation itself. The change in directory feels like it also changes the intended audience, so I think the level of detail needed changes. It's very easy for the docs to go stale if they provide too much detail, as we can see with my recent update that wasn't reflected in the original recovery README.

First stage shuts down containers and cleans up, restoring files from the backup.
Second stage restores cluster database and related files from the backup, relaunching containers and triggering required redeployments as per cluster restore procedure.

Reply:
Rephrased

Reply:
Rephrased